led_strip/include/led_strip_spi.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-FileCopyrightText: 2022-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <stdint.h>
+#include "esp_err.h"
+#include "driver/gpio.h"
+#include "esp_rom_gpio.h"
+#include "driver/spi_master.h"
+#include "driver/spi_common.h"
+#include "hal/spi_types.h"
+#include "led_strip_types.h"
+#include "soc/gpio_sig_map.h"
+#include "soc/spi_periph.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief LED Strip SPI specific configuration
+ */
+typedef struct {
+    rmt_clock_source_t clk_src; /*!< SPI clock source */
```

Comment:
spi_clock_source_t

Because this type was introduced since idf 5.1, so we need to change the dependency of this led_strip component as well